File: exercises/exercism/elixir/nth_prime.exs
defmodule Prime do
  @doc """
  Generates the nth prime.
  """
  @spec nth(non_neg_integer) :: non_neg_integer
  def nth(count) when count < 1, do: raise ArgumentError, message: "count must be equal/greater than 0"
  def nth(count) do
    do_nth(count, [], 2)
  end
  defp do_nth(0, [p | _], _last_pn), do: p
  defp do_nth(count, res, last_pn) do
    if is_prime(last_pn) do
      do_nth(count-1, [last_pn | res], last_pn+1)
    else
      do_nth(count, res, last_pn+1)
    end
  end

  defp is_prime(n) when n <= 1, do: false
  defp is_prime(n) do
    do_is_prime(2, n)
  end
  defp do_is_prime(n, n), do: true
  defp do_is_prime(it, n) do
    if rem(n, it) == 0 do
      false
    else
      do_is_prime(it+1, n)
    end
  end
end

File: exercises/exercism/rust/nth-prime.rs
fn is_prime(n: u32) -> bool {
    if (n == 1) {
        return true;
    }

    for i in 2..n-1 {
        if n % i == 0 {
            return false;
        }
    }

    true
}

pub fn nth(n: u32) -> u32 {
    if (n == 0) {
        return 2;
    }

    let mut counter = 3;
    let mut n_prime = 1;

    loop {
        if is_prime(counter) {
            if n_prime == n {
                break;
            }

            n_prime+=1;
        }
        
        counter+=1;
    }

    counter
}
